classes/acceptances_table.php

```diff
@@ -452,21 +524,46 @@ protected function get_return_url() {
     }
 
     /**
-     * Helper.
+     * Return agreement status
      *
-     * @param int $versionid
+     * @param int $versionid either id of an individual version or empty for overall status
      * @param stdClass $row
      * @return string
      */
     protected function status($versionid, $row) {
-        $status = $row->{'status' . $versionid};
+        $onbehalf = false;
+        $versions = $versionid ? [$versionid => $this->versionids[$versionid]] : $this->versionids; // List of versions.
+        $accepted = []; // List of versionids that user has accepted.
+
+        foreach ($versions as $v => $name) {
+            if (!empty($row->{'status' . $v})) {
+                $accepted[] = $v;
+                $agreedby = $row->{'usermodified' . $v};
+                if ($agreedby && $agreedby != $row->id) {
+                    $onbehalf = true;
+                }
+            }
+        }
+
+        if ($versionid) {
+            $str = new \lang_string($accepted ? 'yes' : 'no');
+        } else {
+            $str = new \lang_string('acceptancecount', 'tool_policy', (object)[
+                'agreedcount' => count($accepted),
+                'policiescount' => count($versions)
+            ]);
+        }
+
         if ($this->is_downloading()) {
-            return empty($status) ? get_string('no') : get_string('yes');
+            return $str->out();
+        } else {
+            $s = $this->output->render(new user_agreement($row->id, $accepted, $this->get_return_url(),
+                $versions, $onbehalf, $row->canaccept));
+            if (!$versionid) {
+                $s .= '<br>' . \html_writer::link(new \moodle_url('/admin/tool/policy/user.php', ['userid' => $row->id]), $str);
```

Comment:
Should we pass `returnurl` to user.php so that we do not end up stuck there?

Reply:
user.php did not have returnurl but I've added it to the template